<launch>
    <arg name="pipeline_path"/>
    <arg name="hydra_ros_ns" default="hydra_ros_node"/>

    <arg name="semantic_map_path" default="''"/>
    <arg name="semantic_label_remap_filepath" default="''"/>
    <arg name="exit_after_clock" default="false"/>
    <arg name="log_path" default="$(find hydra)/output/$(arg dataset_name)"/>
    <arg name="log_timing_incrementally" default="false"/>
    <arg name="disable_timer_output" default="true"/>
    <arg name="timing_disabled" default="false"/>

    <!-- Frontend configuration -->
    <arg name="enable_frontend_output" default="true"/>
    <arg name="enable_places" default="true"/>
    <arg name="bounding_box_type" default="AABB"/>
    <arg name="use_gt_frame" default="true"/>
    <arg name="pose_graph_tracker_type" default="$(eval 'PoseGraphFromOdom' if arg('use_gt_frame') else 'RosPoseGraphs')"/>

    <!-- Backend configuration -->
    <arg name="add_initial_prior" default="true"/>
    <arg name="optimize_on_lc" default="true"/>
    <arg name="use_zmq_interface" default="true"/>
    <arg name="zmq_ip" default="127.0.0.1"/>
    <arg name="zmq_send_url" default="tcp://$(arg zmq_ip):8001"/>
    <arg name="zmq_recv_url" default="tcp://$(arg zmq_ip):8002"/>
    <arg name="zmq_send_mesh" default="true"/>

    <!-- LCD pipeline configuration -->
    <arg name="enable_dsg_lcd" default="false"/>
    <arg name="lcd_use_bow_vectors" default="true"/>
    <arg name="use_gnn_descriptors" default="false"/>
    <arg name="lcd_gnn_object_model" default="$(find hydra)/models/lcd/object_gnn.onnx"/>
    <arg name="lcd_gnn_places_model" default="$(find hydra)/models/lcd/place_gnn.onnx"/>
    <arg name="lcd_gnn_label_embeddings" default="''"/>

    <!-- Openset feature configuration -->
    <arg name="use_openset_features" default="false"/>
    <arg name="feature_receiver_type" value="$(eval 'FeatureReceiver' if arg('use_openset_features') else '&quot;&quot;')"/>

    <rosparam file="$(arg pipeline_path)" subst_value="true" ns="$(arg hydra_ros_ns)"/>
</launch>
